vowpalwabbit/cb_adf.cc

```diff
@@ -506,9 +506,8 @@ base_learner* cb_adf_setup(options_i& options, vw& all)
     return nullptr;
 
   // Ensure serialization of this option in all cases.
-  if (!options.was_supplied("cb_type"))
+  if (options.ensure_default_dependency("cb_type", type_string))
```

Comment:
This is weird, shouldn't cb_type be set with a default value above and we could remove this line

Reply:
Yeah we could actually just set `type_string` instead of running the parse again - then we can also make `ensure_default_dependency` void